ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/AbstractTaskFragment.kt

```diff
@@ -125,14 +126,14 @@ abstract class AbstractTaskFragment<T : AbstractTaskViewModel> : AbstractFragmen
       .showIfTrue(viewModel.isTaskOptional())
 
   private fun onSkip() {
-    if (viewModel.responseText.value.isNullOrEmpty()) {
+    if (viewModel.taskData.value?.orElseGet { null }?.isNullOrEmpty() != false) {
```

Comment:
Can we use `taskDataFlow` instead? `taskData` uses LiveData which we are trying to slowly phase out.

Reply:
I tried using the taskDataFlow, but for some reason I couldn't get it to work. I don't know if there's some issue with the flow not being updated properly or what.

Reply:
You can use `taskDataFlow.value != null` to check whether data is present or not.

Reply:
It'd be nice to add a helper method to the viewModel to add some level of abstraction.

```
fun hasData(): Boolean = taskDataFlow.value != null
```